src/main/java/com/dev/cinema/controller/OrderController.java

```diff
@@ -36,15 +38,25 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Long id) {
-        User user = userService.get(id);
+    public void completeOrder(@RequestParam Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
+            userDetails = (UserDetails) principal;
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
         orderService.completeOrder(shoppingCart);
     }
 
     @GetMapping
-    public List<OrderDtoResponse> getOrderHistory(@RequestParam Long id) {
-        User user = userService.get(id);
+    public List<OrderDtoResponse> getOrderHistory(@RequestParam Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
+            userDetails = (UserDetails) principal;
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
```

Comment:
Maybe better throw your own exception?

Reply:
Seems to me that it was mentioned today. There is no need here in orElseThrow(), we can leave get() here.